### PR TITLE
Plot branches with new arguments 

### LINKED ISSUE #144: Separate line and link arguments in plot
For new users, it is inconvenient and not very clear how to use the `line_colors` and `line_widths` argument in the plot function, when links or transformers are also plotted. It would be better to have separate arguments `line_colors` only for lines and `link_colors` for links and `transformer_colors`, same for `*_widhts`. 

**Optional**
One could have a `branch_colors` as an argument for containing colors for all branches in a mutliindexed Series

It is very hard to keep the backward compatibility for this change and might lead to some breaking cases...

### PR DESCRIPTION
**Pull request by FabianHofmann**
Closes #144 

## Changes proposed in this Pull Request
See issue

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

## Test 

Code was tested with the following 'tests', note that the last one **should** raise an error

```python
import pandas as pd
import matplotlib.pyplot as plt
import networkx as nx
import pypsa

n = pypsa.Network(pypsa.__path__[0] + '/../examples/ac-dc-meshed/ac-dc-data')
n.lopf()
n.plot()
plt.show()

n.plot(line_colors=pd.Series(range(7), n.lines.index), line_cmap='Blues',
       link_widths=2, link_colors='orange', flow='mean')
plt.show()

n.plot(line_colors='green', link_widths=2, link_colors='k', flow='mean')
plt.show()

n.plot(line_colors='green', link_widths=2, link_colors='orange', flow=n.snapshots[0])
plt.show()

n.plot(line_colors='green', link_widths=2, link_colors=pd.Series(range(4), n.links.index),
       link_cmap='Blues', flow=n.snapshots[0])
plt.show()


n.plot(geomap=False)
plt.show()

n.plot(line_colors=pd.Series(range(7), n.lines.index), line_cmap='Blues',
       link_widths=2, link_colors='orange', flow='mean', geomap=False)
plt.show()

n.plot(line_colors='green', link_widths=2, link_colors='k', flow='mean', geomap=False)
plt.show()

n.plot(line_colors='green', link_widths=0, link_colors='orange', flow=n.snapshots[0]
       , geomap=False)
plt.show()

n.plot(line_colors='green', link_widths=.1, link_colors=pd.Series(range(4), n.links.index),
       link_cmap='Blues', flow=n.snapshots[0], geomap=False)
plt.show()


n.plot(line_colors='green', link_widths=.01, link_colors=pd.Series(range(4), n.links.index),
       link_cmap='Blues', flow=n.snapshots[0], geomap=False, line_widths=.001, layouter=nx.layout.circular_layout)
plt.show()


# This should raise an error
n.plot(line_widths=n.branches().length)

```

### CODE REVIEW

File: doc/release_notes.rst [hunk 1]
```diff
@@ -47,6 +47,13 @@ Upcoming Release
 * Examples jupyter notebooks are now available through `binder <https://mybinder.org/>`_.
 * Minor adjustments for compatibility with pandas v1.0.0.
 * After optimizing, the network has now an additional attribute ``objective_constant`` which reflects the capital cost of already existing infrastructure in the network referring to ``p_nom`` and ``s_nom`` values. 
+* Plot function ``n.plot()``, **breaking changes**:
+
+    * New arguments were introduced to the n.plot():  ``link_colors``, ``link_widths``, ``transformer_colors``, ``transformer_widths``, ``link_cmap``, ``transformer_cmap``
+    * ``line_widhts`` and ``line_colors`` now only apply for lines, passing a series with pandas.MultiIndex will raise an error.
+    * ``directed_flow()`` takes only a series with single index now. This enables to use the function within the loop over branch_components.
```
Comment (fneum) [inline doc/release_notes.rst]:
```suggestion
* Changes in the plotting function ``n.plot()`` include some **breaking changes**:

    * A set of new arguments were introduced to separate style parameters of the different branch components:  ``link_colors``, ``link_widths``, ``transformer_colors``, ``transformer_widths``, ``link_cmap``, ``transformer_cmap``
    * ``line_widths``, ``line_colors``, and ``line_cmap`` now only apply for lines and can no longer be used for other branch types (links and transformers). Passing a pandas.Series with a pandas.MultiIndex will raise an error.
    * The function ``directed_flow()`` now takes only a pandas.Series with single pandas.Index run within the loop over individual branches. 
    * The default colours changed.
```

**Comment by FabianHofmann** [comment]
Yes, it would be consistent to make the same changes to `iplot()`, you're right, I'll do it right away. The widths of the flows can be varied with the *_widths arguments. Most of the times they have to be adjusted anyway, a standard size is hard to pick.

**Comment by FabianHofmann** [comment]
Alright, `n.iplot()` is adjusted now. Test with

```python 
from plotly.offline import plot
plot(n.iplot(mapbox=True, line_widths=5, link_widths=2, bus_cmap='Blues',
             bus_colorbar={'title' : 'my colorbar'},
             bus_colors=pd.Series(range(len(n.buses)), n.buses.index),
             bus_alpha=0.9, bus_sizes=20))

```
(I just put all available option in there)

File: pypsa/plot.py [hunk 2]
```diff
@@ -562,27 +559,37 @@ def iplot(n, fig=None, bus_colors='blue',
...
+    link_widths : dict/pandas.Series
+        Widths of links, defaults to 1.5
+    transformer_widths : dict/pandas.Series
+        Widths of transformer, defaults to 1.5
+    line_text : pandas.Series
+        Text for lines, defaults to line names.
+    link_text : pandas.Series
+        Text for links, defaults to line names.
+    tranformer_text : pandas.Series
+        Text for transformers, defaults to line names.
```
Comment (fneum) [inline pypsa/plot.py]:
```suggestion
    line_text : pandas.Series
        Text for lines, defaults to line names.
    link_text : pandas.Series
        Text for links, defaults to link names.
    tranformer_text : pandas.Series
        Text for transformers, defaults to transformer names.
```
This is what you meant right?

File: doc/release_notes.rst [hunk 3]
```diff
@@ -47,6 +47,14 @@ Upcoming Release
 * Examples jupyter notebooks are now available through `binder <https://mybinder.org/>`_.
 * Minor adjustments for compatibility with pandas v1.0.0.
 * After optimizing, the network has now an additional attribute ``objective_constant`` which reflects the capital cost of already existing infrastructure in the network referring to ``p_nom`` and ``s_nom`` values. 
+* Changes in the plotting function ``n.plot()`` and ``n.iplot()`` include some **breaking changes**:
```
Comment (fneum) [inline doc/release_notes.rst]:
```suggestion
* Changes in the plotting functions ``n.plot()`` and ``n.iplot()`` include some **breaking changes**:
```

File: doc/release_notes.rst [hunk 4]
```diff
@@ -47,6 +47,14 @@ Upcoming Release
...
+
+    * A set of new arguments were introduced to separate style parameters of the different branch components:  ``link_colors``, ``link_widths``, ``transformer_colors``, ``transformer_widths``, ``link_cmap``, ``transformer_cmap``
+    * ``line_widths``, ``line_colors``, and ``line_cmap`` now only apply for lines and can no longer be used for other branch types (links and transformers). Passing a pandas.Series with a pandas.MultiIndex will raise an error.
+    * The function ``directed_flow()`` now takes only a pandas.Series with single pandas.Index.
```
Comment (fneum) [inline doc/release_notes.rst]:
```suggestion
    * Additionally, the function `n.iplot()` has new arguments ``line_text``, ``link_text``, ``transformer_text`` to configure the text displayed when hovering over a branch component.
    * The function ``directed_flow()`` now takes only a pandas.Series with single pandas.Index. 
```

**Comment by FabianHofmann** [comment]
Alright! now only the OK of @nworbmot is missing. You can use the snippets above to see the new features.

**Pull request merged**